Config/BuildSettings.swift

```diff
@@ -133,8 +133,16 @@ final class BuildSettings: NSObject {
         // Official Matrix ones
         "matrix.to": ["/"],
         "www.matrix.to": ["/"],
+        // Client Permalinks (for use with `BuildSettings.clientPermalinkBaseUrl`)
+//        "example.com": ["/"],
+//        "www.example.com": ["/"],
     ]
     
+    // For use in clients that use a custom base url for permalinks rather than matrix.to.
+    // This baseURL is used to generate permalinks within the app (E.g. timeline message permalinks).
+    // Optional String that when set is used as permalink base, when nil matrix.to format is used.
+    // Example value would be "https://www.example.com", note there is no trailing '/'.
+    static let clientPermalinkBaseUrl: String? = nil
```

Comment:
It's not clear to see the difference between `matrixPermalinkPaths` and `clientPermalinkBaseUrl ` with variable name. 
Maybe we should something like `supportedPermalinkPaths` and `permalinkGenerationBaseUrl`.
For the `permalinkGenerationBaseUrl` we should maybe set explicitly "matrix.to" by default to make it clear.

Reply:
Maybe we can just change `matrixPermalinkPaths` -> `supportedPermalinkPaths`?
And the rest is clear enough with the comments?

I have used `clientPermalinkBaseUrl ` as it is the same variable naming that is used on the android client. It also has the same behaviour as android regarding being optional/nil.

Reply:
Android uses 'permalink_supported_hosts' so maybe 'supportedPermalinkHosts' on our side and we are pretty consistent then across clients for understanding it.

Reply:
Yes good idea

Reply:
Does Android have an equivalent for `clientPermalinkBaseUrl`? Otherwise we can keep this name.

Reply:
Yea name is the same on android.